Sources/muterCore/Extensions/StringExtensions.swift

```diff
@@ -67,3 +67,33 @@ extension String {
         return String(self[range])
     }
 }
+
+/// Converts a UTF-8 byte offset to a character offset in the given string.
+///
+/// - Parameters:
+///   - string: The string in which the offset needs to be converted.
+///   - utf8Offset: The UTF-8 byte offset to be converted.
+/// - Returns: The corresponding character offset if the conversion is successful, or `nil` if the offset is out of range.
+///
+/// - Complexity: O(*n*), where *n* is the number of characters in the string.
+func convertUtf8OffsetToCharOffset(in string: String, at utf8Offset: Int) -> Int? {
+    if string.utf8.count == string.count { return utf8Offset }
```

Comment:
Reminder that `String.count` is O(*n*). Does that matter for your big-O assessment? I guess it's still linear, but just making sure this is the best way to do it, since you still end up iterating over the string just to do the check.

Also, I'd probably use a `guard` for something like this.

Reply:
@ZevEisenberg Thanks for reminding me. I forgot that `String.count` is O(n).
I've removed that logic